internal/server/server.go

```diff
@@ -0,0 +1,40 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Pigmice2733/peregrine-backend/internal/config"
+	ihttp "github.com/Pigmice2733/peregrine-backend/internal/http"
+	routes "github.com/Pigmice2733/peregrine-backend/internal/routes/v1"
+	"github.com/gorilla/mux"
+)
+
+// Server holds information neccesary for the peregrine-backend server.
+type Server struct {
+	router *mux.Router
+	addr   string
```

Comment:
It's a bit strange to call the TCP address "addr" in one place, but "listen" elsewhere. I would suggest making it the same everywhere, using either "port" or "address".

Reply:
:+1: